app/code/core/Mage/CatalogRule/Model/Rule.php

```diff
@@ -318,6 +318,7 @@ public function applyAll()
         $this->_getResource()->applyAllRules();
         $this->_invalidateCache();
         $indexProcess = Mage::getSingleton('index/indexer')->getProcessByCode('catalog_product_price');
+        $indexProcess->changeStatus(Mage_Index_Model_Process::STATUS_REQUIRE_REINDEX);
```

Comment:
getProcessByCode('catalog_product_price') can return `false` so in order to avoid fatal error we need to call this method inside if statement